src/search/mod.rs

```diff
@@ -0,0 +1,96 @@
+use std::collections::HashMap;
+
+use strum::Display;
+
+#[derive(Debug, Display, PartialEq, Eq, Hash)]
+#[strum(serialize_all = "snake_case")]
```

Comment:
I think the `SearchFilter` enum should be put into the `rspotify-model` crate, then we don't need to impose a new crate and `SearchFilter` is more like a model.